funcframework/framework.go

```diff
@@ -141,6 +141,9 @@ func Start(port string) error {
 			handler = server
 		}
 	}
+	if handler == nil {
```

Comment:
I think we can just `else` (on [`GetRegisteredFunction`](https://github.com/GoogleCloudPlatform/functions-framework-go/pull/138/files#diff-9e12afb8de93767696fd66b5f656f07fc605638967823218b73d4de72a383600R128)) if we didn't get a registered function and state an error.

---

I don't think there can be a case where `handler` will be nil, so I don't think this this case will ever happen. If we can trigger with a test, that would verify this error case is used.

Reply:
For non-declarative functions, the target variable will not be set. If we use an else block, we might break for non-declarative functions. I assume the handler for non-declarative functions will be set in RegisterHTTPFunctionContext. Correct me if I am missing anything.